charmcraft.yaml

```diff
@@ -0,0 +1,123 @@
+# This file configures Charmcraft.
+# See https://juju.is/docs/sdk/charmcraft-config for guidance.
+
+name: takahe-k8s
+type: charm
+title: takahē operator for Kubernetes
+summary: Takahē is an ActivityPub server designed for efficient use on small- to medium-size installations.
+description: |
+  Takahē is an ActivityPub server for small- to medium-size installations, allowing multiple domains on the same infrastructure.
+
+  This charm manages installation and management of a Takahē instance.
+  TODO: Add more details here.
+
+  Takahē can interact with other ActivityPub servers in the 'Fediverse', such as Mastodon,
+  Lemmy, micro.blog, or PeerTube. Takahē is fairly straightforward to install and manage,
+  but this charm simplifies things further, allowing easy deployment and management with
+  Juju, whether on a local Microk8s, a bare metal server, or private or public cloud.
+
+  This charm is intended for use by anyone wishing to manage a Takahē instance, generally
+  for their own private use, with one or more domains.
+
+config:
+  options:
+    main-domain:
+      description: the domain name that will be used for default links (such as in emails). Does not need to be the same as any domain you are hosting user accounts on. Must be provided.
+      type: string
+    media-uri:
+      default: local://
+      description: URI where media should be stored - may be 'local://', 's3://access-key:secret-key@endpoint-url/bucket-name', or 'gcs://bucket-name'
+      type: string
+
+actions:
+  # The wordpress-k8s charm allows deployment with an `initial_settings` config
+  # containing `admin_password`, and has a `get-initial-password` action. Maybe
+  # it would be worth copying that pattern?
+  add-superuser:
+    description: >
+      Add a new superuser account to the instance.
+    params:
+      username:
+        description: Username for the new superuser
+        type: string
+      email:
+        description: Email for the new superuser
+        type: string
+    additionalProperties: false
+    required:
+      - username
+      - email
+  cycle-secret-key:
+    description: >
+      Generate a new internal secret key - this invalidates all sessions.
+      Users will be forced to log in again. This might be useful under security
+      breach circumstances, but should also be done on a regular cadence.
+    additionalProperties: false
+
+bases:
+  - build-on:
+    - name: ubuntu
+      channel: "22.04"
+    run-on:
+    - name: ubuntu
+      channel: "22.04"
+
+peers:
+  takahe-peer:
+    interface: takahē_peers
```

Comment:
> Is this interface name referenced anywhere? Bold using Unicode `ē` in this context! :-) Per voice discussion, I'd probably remove it from here (but keeping it in the Python charm code seems reasonable).

As per voice conversation, I did want to figure out where I could use a macron without things breaking - mostly being curious about how Juju and ops support for languages outside of English is (not great, as it turns out! Charm and model names give an error saying that they have to be "letters" without saying that it's actually ASCII/Latin letters, containers and images made deploying hang, and a relation endpoint junked my whole controller!). I couldn't use `ē` for the GitHub Repo name either, for what it's worth :disappointed:.

I agree that now that I've discovered what it's like, it's likely best to stick to ASCII for everything in the metadata including the interface name.

> Also, per https://juju.is/docs/sdk/styleguide#heading--naming, should we use hyphens rather than underscores for these names? (Similar with `postgresql_client` below)

Ah, I missed that, thanks! Will change. The k8s tutorial gets this 'wrong' by the way (both for peer and postgres). I don't feel it's worth opening ~11 PRs to change that, though...

Reply:
Ah, also it's not actually used anywhere - does the interface name have any real meeting for a peer relation? It seems like it's there because non-peer relations need an interface, but the peers don't really need to be matched up to a suitable interface. As far as I know, you can only have a single peer relation, so it can't be used for distinguishing between multiple.